google/cloud/aiplatform/docker_utils/run.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import os
+from pathlib import Path
+import re
+from typing import Dict, List, Optional, Sequence
+
+try:
+    import docker
+except ImportError:
+    raise ImportError(
```

Comment:
What's stopping this from being part of the package root setup.py?

Reply:
To keep install size small, maybe this doesn't need to be included in setup.py. That said, lets not break import of the rest of the SDK by throwing when docker is missing.

Lets try one of the following:
1. Importing this module only when it's strictly required for usage 
2. Moving the try/except import call to within the method using it
3. Doing a "try: import docker except docker=None" and then at invocation time throwing with an informative exception if docker is None.

Reply:
I think users won't get this exception while they don't use any classes we added for CPR because we don't import it in the root of the SDK?
If users want to use CPR related classes, they can install with "google-cloud-aiplatform[prediction]" additionally.